### PR TITLE
Added Viu

### PR DESCRIPTION
**Pull request by farizrifqi**
image
image
image
image
image
image
image
image
image
image
image
image
image
image

### CODE REVIEW

File: websites/V/Viu/presence.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    browsing: "presence.activity.browsing"
+});
+function getTimestamps(
+  videoTime: number,
+  videoDuration: number
+): Array<number> {
+  var startTime = Date.now();
+  var endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+  return [Math.floor(startTime / 1000), endTime];
+}
```
Comment (Bas950) [inline websites/V/Viu/presence.ts:16]: use presence.getTimestamps

**Review by Bas950 — CHANGES_REQUESTED** [review]
Fix codacy issues, and remove the files in the main V folder.

File: websites/V/Viu/presence.ts (hunk 1 above)
Comment (farizrifqi) [reply websites/V/Viu/presence.ts]: done, all fixed

File: websites/V/Viu/presence.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+			  : (await strings).play),
+			(presenceData.startTimestamp = timestamps[0]),
+			(presenceData.endTimestamp = timestamps[1]);
+		if (video.paused) {
+			delete presenceData.startTimestamp;
+			delete presenceData.endTimestamp;
+		}
+	}else{
+		var browsingStamp = Math.floor(Date.now() / 1000);
+		presenceData.startTimestamp = browsingStamp;
```
Comment (Bas950) [inline websites/V/Viu/presence.ts:44]: no var

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/V/Viu/presence.ts (hunk 2 above)
Comment (farizrifqi) [reply websites/V/Viu/presence.ts]: changed

File: websites/V/Viu/presence.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+			}
+			presenceData.state = "Trailer";
+		}else{
+			presenceData.details = document.querySelector("h1.ep_title").textContent;
+			presenceData.state = "Movie";
+		}
+		(presenceData.smallImageKey = video.paused ? "pause" : "play"),
+			(presenceData.smallImageText = video.paused ? (await strings).pause : (await strings).play),
+			(presenceData.startTimestamp = timestamps[0]),
+			(presenceData.endTimestamp = timestamps[1]);
```
Comment (Bas950) [inline websites/V/Viu/presence.ts:35]: just do ; after each line and not with (),

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/V/Viu/presence.ts (hunk 3 above)
Comment (farizrifqi) [reply websites/V/Viu/presence.ts]: fixed

**Review by Bas950 — APPROVED** [review]

File: websites/V/Viu/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+		"name": "zeraneL",
+		"id": "209204682156015616"
+	},
+	"url": [
+		"viu.com",
+		"www.viu.com"
+	],
+	"description": {
+		"id": "Layanan video over-the-top",
+		"en": "Over-the-top video services"
```
Comment (ririxidev) [inline websites/V/Viu/dist/metadata.json:13]: fr?

File: websites/V/Viu/presence.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+		}else if (document.location.pathname.includes("playlist-")) {
+			presenceData.details = "Viewing Page";
+			presenceData.state = document.querySelector("h1.headerContainer__info__title").textContent;
+		}else if (document.location.pathname.includes("collection-")) {
+			presenceData.details = "Viewing Collection";
+			presenceData.state = document.querySelector("h1.c-headingtitle").textContent;
+		}
+	}
+	if (presenceData.details == null) {
+		presence.setTrayTitle();
```
Comment (ririxidev) [inline websites/V/Viu/presence.ts:78]: why you set TrayTitle to null if you don't even set it?

**Review by ririxidev — CHANGES_REQUESTED** [review]

File: websites/V/Viu/dist/metadata.json (hunk 4 above)
Comment (farizrifqi) [reply websites/V/Viu/dist/metadata.json:13]: required in fr version?

Comment (ririxidev) [reply websites/V/Viu/dist/metadata.json:13]: No. Fr is For real? Change these description to something better....

Comment (Bas950) [reply websites/V/Viu/dist/metadata.json:13]: Thats the official description.

**Comment by ririxidev** [comment]
No response from pull request author, then we are closing this pull request. If you will want to re-do this pull request, feel free to do!

**Pull request closed without merging**